libraries/botbuilder-core/botbuilder/core/bot_state.py

```diff
@@ -24,8 +25,7 @@ def is_changed(self) -> bool:
         return self.hash != self.compute_hash(self.state)
 
     def compute_hash(self, obj: object) -> str:
-        # TODO: Should this be compatible with C# JsonConvert.SerializeObject ?
-        return str(obj)
+        return str(Pickler().flatten(obj))
```

Comment:
Great catch! This is so important that I'm surprised that didn't blow up earlier